### PR TITLE
Release v0.0.2

### PR DESCRIPTION
**Pull request by briskt**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+func readFlags() (params AppConfigParams) {
+	flag.StringVar(&params.applicationID, "app", "", "application identifier")
+	flag.StringVar(&params.environmentID, "env", "", "environment identifier")
+	flag.StringVar(&params.configProfileID, "config", "", "config profile identifier")
+	flag.BoolVar(&verbose, "v", false, "verbose output")
+	flag.Parse()
+
+	if params.applicationID == "" {
+		log.Printf("specify application identifier with --app flag")
+		os.Exit(1)
```
Comment (jason-jackson) [inline main.go]:
```suggestion
		log.Fatalf("specify application identifier with --app flag")
```

**Review by jason-jackson — COMMENTED** [review]
I'd need to take a closer look at config, but go code looks fine, could replace the log + os.Exit(1) by Fatatf if I remember correctly. I will look again tomorrow

File: main.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+func readFlags() (params AppConfigParams) {
+	flag.StringVar(&params.applicationID, "app", "", "application identifier")
+	flag.StringVar(&params.environmentID, "env", "", "environment identifier")
+	flag.StringVar(&params.configProfileID, "config", "", "config profile identifier")
+	flag.BoolVar(&verbose, "v", false, "verbose output")
+	flag.Parse()
+
+	if params.applicationID == "" {
+		fmt.Printf("specify application identifier with --app flag")
```
Comment (jason-jackson) [inline main.go]: Don't strictly need Printf here, but it's fine.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	if params.environmentID == "" {
+		fmt.Printf("specify environment identifier with --env flag")
+		os.Exit(1)
+	}
+	if params.configProfileID == "" {
+		fmt.Printf("specify config profile identifier with --config flag")
+		os.Exit(1)
+	}
+
+	if flag.Arg(0) == "" {
```
Comment (jason-jackson) [inline main.go]: I think I like this more, but could keep it if you want.

```suggestion
	if flag.NArg() == 0 {
```

File: main.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	fmt.Printf("read %d lines from AppConfig", len(vars))
+	if verbose {
+		fmt.Printf("vars: %s", vars)
+	}
+
+	return vars, nil
+}
+
+func getVars(config string) []string {
+	lines := strings.Split(config, "\n")
```
Comment (jason-jackson) [inline main.go:112]: An alternative would be to parse it as csv (using encoding/csv). No real preference, the csv would handle the quotes, but could have unintended side effects (like what should happen with double quotes?).

Also, if it's in bytes, you could just pass that and use a byte reader instead.

```
	r := csv.NewReader(strings.NewReader(config))
	r.Comma = '='
	r.Comment = '#'
	r.FieldsPerRecord = 2 // not sure if this causes an error if not matching or if skips, use at own risk

	records, _ := r.ReadAll()
	result := make([]string, len(records))
	for i, v := range records {
		result[i] = strings.Join(v, "=")) // or use string(r.Comma)
	}
	return result
```

Or if `r.FieldsPerRecord` doesn't work:
```
	r := csv.NewReader(strings.NewReader(config))
	r.Comma = '='
	r.Comment = '#'

	records, _ := r.ReadAll()
	var result []string
	for i, v := range records {
		if len(v) == 2 {
			result = append(result, strings.Join(v, "="))
		}
	}
	return result
```

**Review by jason-jackson — APPROVED** [review]

File: main.go (hunk 2 above)
Comment (briskt) [reply main.go]: True. Your comment also made me realize that I lost the newlines when I went to `Printf`.

File: main.go (hunk 4 above)
Comment (briskt) [reply main.go:112]: Interesting idea, but I am not sure about the added complexity and uncertainty that brings.

**Pull request merged**